### PR TITLE
DEV: Update ember-template-lint

### PR DESCRIPTION
**Pull request by CvX**

### CODE REVIEW

File: .template-lintrc.js
Comment (CvX) [inline .template-lintrc.js:1]: Temporarily disabled new (offending) rules

File: app/assets/javascripts/discourse/app/components/d-toggle-switch.hbs [hunk 1]
```diff
@@ -1,14 +1,14 @@
 <div class="d-toggle-switch">
   <label class="d-toggle-switch--label">
-    {{! template-lint-disable no-unnecessary-concat }}
+    {{! template-lint-disable no-redundant-role }}
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/d-toggle-switch.hbs:3]: it was (incorrectly I believe) complaining about the use of `role="switch"` on a `<button>`

File: app/assets/javascripts/discourse/app/components/d-toggle-switch.js [hunk 2]
```diff
@@ -2,14 +2,15 @@ import Component from "@glimmer/component";
 import { tracked } from "@glimmer/tracking";
 import I18n from "I18n";
 
-export default class DiscourseToggleSwitch extends Component {
-  @tracked iconEnabled = true;
-  @tracked showIcon = this.iconEnabled && this.icon;
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/d-toggle-switch.js:7]: Those were unused

File: app/assets/javascripts/select-kit/addon/components/period-chooser/period-chooser-header.hbs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<h2 class="selected-name" title={{this.title}} role="option">
+<h2 class="selected-name" title={{this.title}}>
```
Comment (CvX) [inline app/assets/javascripts/select-kit/addon/components/period-chooser/period-chooser-header.hbs:1]: `option` is not a valid role in all three of these places - they're not selectable options (they select-kit headers) and don't have aria-selected attr

**Review by keegangeorge — APPROVED** [review]
Nice!

**Pull request merged**